### PR TITLE
Basic plugin dev structure

### PR DESCRIPTION
**Pull request by goldenapples**
Adds .editorconfig, composer setup including WP Coding Standards, and a
grunt task to run WP Coding Standards sniffs against every save.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "fusioneng/image-shortcake",
+  "authors": [
+    {
+      "name": "Nathaniel Taintor",
+      "email": "goldenapplesdesign@gmail.com"
```
Comment (danielbachhuber) [inline composer.json]: Can we make this Fusion Engineering and tech@fusion.net?

File: Gruntfile.js [hunk 2]
```diff
@@ -27,6 +27,23 @@ module.exports = function( grunt ) {
 			},
 		},
 
+		phpcs: {
+			plugin: {
+				src: [ '*.php' ],
```
Comment (danielbachhuber) [inline Gruntfile.js]: Does this work for files in subdirectories too ? e.g. recursively

**Comment by danielbachhuber** [comment]
Couple of questions

File: Gruntfile.js (hunk 2 above)
Comment (goldenapples) [inline Gruntfile.js]: No, it doesn't. I think the better way of doing this is setting `extensions: "php"` and then specify all the directories not to check with the `ignore` parameter. I wasn't sure at first what those ignores would be, but ignoring `*/tests/*,*/vendor/*` should be enough for starts.

Comment (danielbachhuber) [inline Gruntfile.js]: :+1: — although I wouldn't mind if `*/tests/*` met coding standards.

**Comment by goldenapples** [comment]
Fixed phpcs task & changed author name. Back to you.

**Pull request merged**